akka-stream/src/main/scala/akka/stream/scaladsl/RestartFlow.scala

```diff
@@ -399,7 +403,7 @@ object RestartWithBackoffFlow {
 
         override protected def onTimer(timerKey: Any): Unit = {
           log.debug(s"Stage was canceled after delay of $delay")
-          completeStage()
+          cancelStage(cause.get)
```

Comment:
`cause` will always be `Some` because the timer was started from onDownstreamFinish, but feels somewhat future unsafe. Maybe match and complete versus cancel?

Reply:
I think it's fine. It's the main function of the whole stage to do exactly this. If this basic invariant is not guaranteed, it's fine to fail as hard as possible IMO.

Reply:
I added an extra clear exception here instead.